Documentation/Storage-Configuration/Ceph-CSI/ceph-csi-volumegroup-snapshot.md

```diff
@@ -0,0 +1,107 @@
+refer to VolumeGroupSnapshot documentation
+[here](https://github.com/kubernetes-csi/external-snapshotter/tree/master#volume-group-snapshot-support) for more details.
+
+- We also need a `VolumeGroupSnapshotClass` for volume group snapshot to work. The purpose of a `VolumeGroupSnapshotClass` is
+defined in [the kubernetes
+documentation](https://kubernetes.io/blog/2023/05/08/kubernetes-1-27-volume-group-snapshot-alpha/).
+In short, as the documentation describes it:
+
+!!! info
+    Created by cluster administrators to describe how volume group snapshots
+    should be created. including the driver information, the deletion policy, etc.
+
+## CephFS Snapshots
+
+### VolumeGroupSnapshotClass
+
+In [VolumeGroupSnapshotClass](https://github.com/rook/rook/tree/master/deploy/examples/csi/cephfs/groupsnapshotclass.yaml),
+the `csi.storage.k8s.io/group-snapshotter-secret-name` parameter should reference the
+name of the secret created for the cephfs-plugin.
+
+In the `VolumeGroupSnapshotClass`, update the value of the `clusterID` field to match the namespace
+that Rook is running in. When Ceph CSI is deployed by Rook, the operator will automatically
+maintain a configmap whose contents will match this key. By default this is
+"rook-ceph".
+
+```console
+kubectl create -f deploy/examples/csi/cephfs/groupsnapshotclass.yaml
+```
+
+### CephFS VolumeGroupSnapshot
+
+In [VolumeGroupSnapshot](https://github.com/rook/rook/tree/master/deploy/examples/csi/cephfs/groupsnapshot.yaml),
+`volumeGroupSnapshotClassName` should be the name of the `VolumeGroupSnapshotClass`
+previously created. The labels inside `matchLabels` should be present on the
+PVCs that are already created by the CephFS CSI driver.
+
+```console
+kubectl create -f deploy/examples/csi/cephfs/groupsnapshot.yaml
+```
+
+### Verify CephFS GroupSnapshot Creation
+
+```console
+$ kubectl get volumegroupsnapshotclass
+NAME                              DRIVER                          DELETIONPOLICY   AGE
+csi-cephfsplugin-groupsnapclass   rook-ceph.cephfs.csi.ceph.com   Delete           21m
+```
+
+```console
+$ kubectl get volumegroupsnapshot
+NAME                       READYTOUSE   VOLUMEGROUPSNAPSHOTCLASS          VOLUMEGROUPSNAPSHOTCONTENT                              CREATIONTIME   AGE
+cephfs-pvc-groupsnapshot   true         csi-cephfsplugin-groupsnapclass   groupsnapcontent-d13f4d95-8822-4729-9586-4f222a3f788e   5m37s          5m39s
+```
+
+The snapshot will be ready to restore to a new PVC when `READYTOUSE` field of the
+`volumegroupsnapshot` is set to true.
+
+### Restore the CephFS groupsnapshot to a new PVC
+
+We need to find the name of the snapshots created by the `VolumeGroupSnapshot` first by running:
+
+```console
+$ kubectl get volumegroupsnapshot/cephfs-pvc-groupsnapshot -o=jsonpath='{range .status.pvcVolumeSnapshotRefList[*]}PVC: {.persistentVolumeClaimRef.name}, Snapshot: {.volumeSnapshotRef.name}{"\n"}{end}'
```

Comment:
please add namespace flag to the CLI

Reply:
We do not use explicit namespace in any of the other resources.